### DISCUSSION TITLE
Why I'm getting `undefined` instead of `edge` for this `initialData` value?

### DISCUSSION THREAD

**Discussion by frederikhors** [TanStack/query#6037] [Q&A]
Using the code below I'm getting always `undefined` as the `initialData` value, why? I expect to get `edge` instead.

Obviously in the cache are present the `edges` for `playerKeys.lists()` (from the list view).

```ts
//... player_id comes from here

const playerStore = createQuery({
  queryKey: playerKeys.byId(player_id),
  queryFn: playerById(player_id),
  initialData: () => {
    // I'm using getQueriesData not getQueryData here
    const lists = queryClient.getQueriesData<playerListConnection>(
      playerKeys.lists()
    );

    return lists.find((query) => {
      return query[1]?.edges.find((edge) => {
        return edge.id === player_id;
      });
    });
  },
});
```

**Comment by TkDodo** [comment]
the answer from captain obvious would be: Because your `initialData` function returns `undefined`. So, it's impossible to say from just seeing this code snippet. Post a runnable reproduction please.
